pinot-core/src/main/java/com/linkedin/pinot/core/io/reader/impl/v1/BaseChunkSingleValueReader.java

```diff
@@ -105,17 +140,21 @@ protected ByteBuffer getChunkForRow(int row, ChunkReaderContext context) {
       chunkSize = nextChunkOffset - chunkPosition;
     }
 
-    ByteBuffer uncompressedBuffer = context.getChunkBuffer();
-    uncompressedBuffer.clear();
+    ByteBuffer decompressedBuffer = context.getChunkBuffer();
+    decompressedBuffer.clear();
 
     try {
-      _chunkDecompressor.decompress(_dataBuffer.toDirectByteBuffer(chunkPosition, chunkSize), uncompressedBuffer);
+      if (_chunkDecompressor != null) {
+        _chunkDecompressor.decompress(_dataBuffer.toDirectByteBuffer(chunkPosition, chunkSize), decompressedBuffer);
+      } else {
+        decompressedBuffer.put(_dataBuffer.toDirectByteBuffer(chunkPosition, chunkSize));
```

Comment:
Do we need the copy here? can we return a slice of the dataBuffer here?

Reply:
Yes, it does not work otherwise, because the decompressed buffer in context is not updated, the subsequent reads don't get the data. I had thought about of an option to overwrite the chunkBuffer in the context, but that would require making it mutable, and didn't like it. If a client sets chunkBuffer in the context, I couldn't convince myself to over-write it under the hood.